### PR TITLE
Add test for fate#320597 'zypper lifecycle'

### PR DESCRIPTION
**Pull request by okurz**
Verification run: http://lord.arch/tests/2449

### CODE REVIEW

File: tests/console/zypper_lifecycle.pm [hunk 1]
```diff
@@ -0,0 +1,66 @@
+# SUSE's openQA tests
+#
+# Copyright © 2009-2013 Bernhard M. Wiedemann
+# Copyright © 2012-2016 SUSE LLC
```
Comment (coolo) [inline tests/console/zypper_lifecycle.pm]: does bernhard know?

Comment (okurz) [inline tests/console/zypper_lifecycle.pm]: done

File: products/sle/main.pm [hunk 2]
```diff
@@ -572,6 +572,7 @@ sub load_consoletests() {
         loadtest "console/vim.pm";
         if (!is_staging()) {
             loadtest "console/firewall_enabled.pm";
+            loadtest "console/zypper_lifecycle.pm";
```
Comment (coolo) [inline products/sle/main.pm]: Does it really have to run in that many scenarios? It seems to be pretty generic test that can easily run in console extra tests

Comment (coolo) [inline products/sle/main.pm]: specifically I'm wondering if this should run for GA maintenance updates too

Comment (okurz) [inline products/sle/main.pm]: ah, ok. should be >=12-SP2 as it's a new feature for that version but as it's a feature validation test I understood it should not just be part of extratests.

Comment (coolo) [inline products/sle/main.pm]: extratests is just stupidly named, but it's a pool of tests where the scenario doesn't matter. Having it in regular console tests is important if storage or software selection makes a difference. This doesn't seem to be the case with this.

Comment (sysrich) [inline products/sle/main.pm]: software selection has an impact on this test, as additional addons should produce an output for those products

This is especially important for modules, which have a different lifecycle from the rest of the product (the whole point of this feature)

**Comment by okurz** [comment]
now still in consolete tests but '>= 12-SP2'

**Comment by coolo** [comment]
So in the call this morning we agreed to introduce a new variable to run additional console tests that need more coverage but not so much to run in every scenario.

This needs to be set in one test suite run per product and main.pm can calculate it to run when we have ADDONS or other things.

**Comment by lnussel** [comment]
I'd also recommend to try putting the logic into a standalone script in the data directory rather than coding everything into openQA. The script could be run manually outside openQA then which makes debugging easier. openQA should only evaluate the result.

**Comment by okurz** [comment]
@coolo yes, can I do this after RC1 please?

@lnussel I don't agree. The test code should be pretty straight forward easily readable. I am not coding it into openQA rather a simple perl test module and we can call os-autoinst also on non-virtual machines. Putting it into an external script would make debugging harder when run from openQA context. I would agree if you were talking about a CI system like jenkins but for openQA it makes more sense to have the actions visible from within the system. I would be happy to support in pushing jenkins for system testing but this is unrelated.

**Comment by sysrich** [comment]
Merging on the condition that https://progress.opensuse.org/issues/13040 will be handled in the near future

File: products/sle/main.pm [hunk 3]
```diff
@@ -622,9 +622,12 @@ sub load_consoletests() {
         if (get_var("CLONE_SYSTEM")) {
             loadtest "console/yast2_clone_system.pm";
         }
-        if (check_var('ARCH', 'aarch64') and check_var('VERSION', '12-SP2')) {
+        if (check_var('ARCH', 'aarch64') and sle_version_at_least('12-SP2')) {
             loadtest "console/check_gcc48_on_sdk_in_aarch64.pm";
         }
+        if (sle_version_at_least('12-SP2')) {
```
Comment (coolo) [inline products/sle/main.pm]: looks like staging projects don't have a life cycle

Comment (okurz) [inline products/sle/main.pm]: yes, shouldn't have been there anyway.

Comment (okurz) [inline products/sle/main.pm]: please see #1650

**Pull request merged**